tools/provision-vms.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+source "common.sh"
+source "$PROJECT_DIR/tools/vbmc-funcs.sh"
+
+CLUSTER_NAME="kni-upi-lab"
+
+NUM_MASTERS=$(yq -r '.controlPlane.replicas' $PROJECT_DIR/cluster/install-config.yaml)
+NUM_WORKERS=$(yq -r '.compute[0].replicas' $PROJECT_DIR/cluster/install-config.yaml)
+
+MASTER_PROV_MAC_PREFIX="52:54:00:82:68:4"
+MASTER_BM_MAC_PREFIX="52:54:00:82:69:4"
+
+WORKER_PROV_MAC_PREFIX="52:54:00:82:68:5"
+WORKER_BM_MAC_PREFIX="52:54:00:82:69:5"
+
+MASTER_VBMC_PORT_START=624
+WORKER_VBMC_PORT_START=625
+
+LIBVIRT_STORAGE_POOL="default"
+
+(
+    $PROJECT_DIR/tools/clean-vms.sh
+) || exit 1
+
+for i in $(seq 0 $((NUM_MASTERS - 1))); do
+    name="$CLUSTER_NAME-master-$i"
+
+    sudo virt-install --ram 16384 --vcpus 4 --os-variant rhel7 --cpu host-passthrough --disk size=40,pool=$LIBVIRT_STORAGE_POOL,device=disk,bus=virtio,format=qcow2 --import --noautoconsole --vnc --network=bridge:provisioning,mac="$MASTER_PROV_MAC_PREFIX$i" --network=bridge:baremetal,mac="$MASTER_BM_MAC_PREFIX$i" --name "$name" --os-type=linux --events on_reboot=destroy,on_lockfailure=poweroff --boot hd,network
```

Comment:
This is fine for now, but as an improvement, we could parametrize the system requirements (RAM, vCPUs, etc) and add those vars to common.sh, so the user with less powerful systems can change the flavor of the VMs.